src/content/blog/2022-08-05-new-config-system-part-2.md

```diff
@@ -0,0 +1,512 @@
+
+The top-level `settings` key behaves the exact same way as in eslintrc. You an define an object with key-value pairs that should be available to all rules. Here's an example:
+
+```js
+export default [
+    {
+        settings: {
+            sharedData: "Hello"
+        }
+    }
+];
+```
+
+### Using predefined configs
+
+ESLint has two predefined configs:
+
+* `eslint:recommended` - enables the rules that ESLint recommends everyone use to avoid potential errors
+* `eslint:all` - enables all of the rules shipped with ESLint
+
+To include these predefined configs, you can insert the string values into the returned array and then make any modifications to other properties in subsequent configuration objects:
+
+```js
+export default [
+    "eslint:recommended",
+    {
+        rules: {
+            semi: ["warn", "always"]
+        }
+    }
+];
+```
+
+Here, the `eslint:recommended` predefined configuration is applied first and then another configuration object adds the desired configuration for `semi`.
+
+## Backwards compatibility utility
+
+As mentioned previously, we felt like there needed to be a good amount of backwards compatibility with eslintrc in order to ease the transition. The [`@eslint/eslintrc`](https://npmjs.com/package/@eslint/eslintrc) package provides a `FlatCompat` class that makes it easy to continue using eslintrc-style shared configs and settings within a flat config file. Here's an example:
+
+```js
+import { FlatCompat } from "@eslint/eslintrc";
+
+const compat = new FlatCompat({
+    baseDirectory: __dirname
+});
+
+export default [
+
+    // mimic ESLintRC-style extends
+    compat.extends("standard", "example"),
+
+    // mimic environments
+    compat.env({
+        es2020: true,
+        node: true
+    }),
+
+    // mimic plugins
+    compat.plugins("airbnb", "react"),
+
+    // translate an entire config
+    compat.config({
+        plugins: ["airbnb", "react"],
+        extends: "standard",
+        env: {
+            es2020: true,
+            node: true
+        },
+        rules: {
+            semi: "error"
+        }
+    })
+];
+```
+
+Using the `FlatCompat` class allows you to continue using all of your existing eslintrc files while optimizing them for use with flat config. We envision this as a necessary transitional step to allow the ecosystem to slowly convert over to flat config.
+
+## Conclusion
+
+The team spent a long time designing flat config so that it would both feel familiar to existing users and to provide new functionality that would benefit everyone. We kept things like rules, settings, and processors the same while extending things like plugins, language options, and linter options to be more uniform. We think that flat config has found a good balance between these two poles and that you will enjoy using ESLint more once the new config system is generally available. In the meantime, the compatibility utility will allow you to continue using existing shared configs.
```

Comment:
```suggestion
The team spent a long time designing flat config so that it would both feel familiar to existing users and provide new functionality that would benefit everyone. We kept things like rules, settings, and processors the same while extending things like plugins, language options, and linter options to be more uniform. We think that flat config has found a good balance between these two poles and that you will enjoy using ESLint more once the new config system is generally available. In the meantime, the compatibility utility will allow you to continue using existing shared configs.
```